api/grpc/interceptor/AuthnInterceptor.go

```diff
@@ -40,8 +44,27 @@ type providerJSON struct {
 	JwksURL string `json:"jwks_uri"`
 }
 
-// NewAuthnInterceptor constructor
-func NewAuthnInterceptor(config api.AuthConfig) (*AuthnInterceptor, error) {
+// NewAuthnInterceptor creates a new AuthnInterceptor
+func NewAuthnInterceptor(configs []api.AuthConfig) (*AuthnInterceptor, error) {
+	if configs == nil {
+		return nil, fmt.Errorf("configs should not be nil")
+	}
+
+	var providers []*authnProvider
```

Comment:
Should we check if the authConfigs are duplicated somewhere around here or even earlier when initializing the config? So when there's a "duplicated entry" with same issuer, jwksuri etc, I don't exactly know how the Cache would behave (e.g. overwrite would be fine I guess) - thinking of some "Set"-like semantics

Reply:
I've been wondering how to handle cases like duplication or nil providers, and thought about adding some logic, but I'm not sure. I think it potentially complicates the code for no real benefit. Either you get what you want anyway, and there's no error, or something fails hard, which is good. In your example, having 2 x identical authnproviders is not really hurting anything, it's just bad config. (But I'm open to being convinced otherwise.)